### ISSUE TITLE
Failed to execute script on Windows 10.

### ISSUE THREAD

**Issue by jfedotov** [pyinstaller/pyinstaller#4040]
Windows 10, python 3.7.2
Exe file has been created with _pyinstaller --onefile  main.py_

But I can't execute it and get the error: 
```
  File "site-packages\urllib3\packages\six.py", line 316, in <module>
  File "site-packages\urllib3\packages\six.py", line 128, in __init__
SystemError: <method-wrapper '__get__' of getset_descriptor object at 0x00183FD0> returned a result with an error set
[2408] Failed to execute script main
```

The same script works fine on Linix and MacOS. Also pure python main.py works fine on Windows10. Executables on linux and macos are created with the same pyinstaller parameters.
I get this issue only with exe on Windows 10.

Looks like some issues with "six" module.
Also I've got same wih "functool"
```
  File "c:\users\efedo\appdata\local\programs\python\python37-32\lib\functools.py", line 54, in update_wrapper
    value = getattr(wrapped, attr)
SystemError: <method-wrapper '__get__' of getset_descriptor object at 0x006AEE90> returned a result with an error set
[12116] Failed to execute script main`
```
If I remove "import six" and "import functool" from main.py and re-run pyinstaller, then exe works fine....

**Comment by htgoebel** [comment]
Please provide more details, see [How to report Bugs](https://github.com/pyinstaller/pyinstaller/wiki/How-to-Report-Bugs). Esp. please provide a *minimal* example to reproduce the error.

Beside this:
> If I remove "import six" and "import functool" from main.py and re-run pyinstaller, then exe works fine....

So this is the workaround :-)

**Comment by jfedotov** [comment]
:) As workaround, I've added SystemError as exception into functool and six at affected lines. Do not know why freezed app crashed while everything is working fine in python env.

**Comment by aleksaa01** [comment]
Something changed in PySide2 5.12.1.. And simply by going from version 5.12.1 down to 5.12.0 you can fix the error(at least for me).
It's weird because only thing that was failing in my case were these two methods from signal.py from standard python library:

```
@_wraps(_signal.signal)
def signal(signalnum, handler):
    handler = _signal.signal(_enum_to_int(signalnum), _enum_to_int(handler))
    return _int_to_enum(handler, Handlers)


@_wraps(_signal.getsignal)
def getsignal(signalnum):
    handler = _signal.getsignal(signalnum)
    return _int_to_enum(handler, Handlers)
```

More specifically, those two decorators were the problem.

**Comment by PierreRochard** [comment]
fwiw this is not windows specific, I had the same issue on macOS and after a lot of fruitless debugging I ended up just pinning the pyside dependency to the last version that worked

**Comment by albertosottile** [comment]
Just seen the same issue on macOS and PySide2 5.12.1, I also had to downgrade to 5.12.0.

**Comment by albertosottile** [comment]
FYI, I reported the issue on the PySide2 tracker:
https://bugreports.qt.io/browse/PYSIDE-942

**Comment by tappi287** [comment]
Thanks for reporting it to the PySide team! From the qt developer feedback, the root cause is shiboken2 not locating it's uncompiled python modules.

This local hook works for me:
```
from PyInstaller.utils.hooks import collect_data_files

datas = collect_data_files('shiboken2', include_py_files=True, subdir='support')
```
Save this as hook-shiboken2.py and add to your --additional-hooks-dir

**Comment by albertosottile** [comment]
I was not aware of the `include_py_files` option, which actually allowed to solve the issue. Thanks for writing this hook, I am going to test it tomorrow. I think that the PySide team is still working on finding a solution that allows pyinstaller (and other installers) to find the `shiboken2.support.signature.loader` module independently.

**Comment by albertosottile** [comment]
@tappi287 I am trying your hook but I could not manage to make it work. The `support` folder was still not copied in the frozen directory. I tried both the `--additional-hooks-dir` argument and the `hookspath=[]` option in the .spec file.

**Comment by tappi287** [comment]
@albertosottile Good to know that it works then, since this was the very first hook I've written.
Btw, you mentioned not working `--additional-hooks-dir` argument. I've actually not tested this. I've added my local hook directory, containing the `hook-shikoken2.py` to the spec file and can confirm this works on Windows 10.
```
my_local_hooks = ['hooks']

# Removed other arguments for this short example
a = Analysis(
             hookspath=my_local_hooks,
)
```

**Comment by ctismer** [comment]
@tappi287 Great that you found that solution, I was too unexperienced to to that.

The solution is not complete. The real one must also include the second support folder:

`from PyInstaller.utils.hooks import collect_data_files

datas = collect_data_files('shiboken2', include_py_files=True, subdir='support')
datas += collect_data_files('PySide2', include_py_files=True, subdir='support')
`

**Comment by strider34** [comment]
Just been through this and the only thing that worked was changing Pyside2 from 5,12,1 to 5.12.0
I am on Windows 10 and Python 3.7.1. Hope that helps.

**Comment by ctismer** [comment]
The above patch works for Python 3.7 and PySide 5.12.1+ .

For PyInstaller, it should be split into two hooks, and PySide2 should be merges with the existing one.

**Comment by ctismer** [comment]
@strider34 That only hides the error. It is still wrong. The signature module must be loaded, or you can wait for crashes.

**Comment by albertosottile** [comment]
@htgoebel Should we open a PR for including the additional hook(s) described above into PyInstaller?

**Comment by ctismer** [comment]
I'm sorry, but there were so many problems with the 5.12.1 release, that I needed to evict this problem, completely. The next version will always work, because I'm embedding all of Python code
as a fallback. The embedding prototype is already working. Await this early next week.

**Comment by bjones1** [comment]
A PR is welcome. What's really needed is a thorough revision of all the PySide 2 hooks, following the PyQt5 hooks (which have the new code).

**Comment by bjones1** [comment]
This was fixed in 5b5791285332003940f4bc70c32bb0cf10a32917. Please reopen if there are still problems.
